lopezMidterm/src/components/Checkout.tsx

```diff
@@ -0,0 +1,97 @@
+import React, { useEffect, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { RootState } from '../store';
+import { removeFromCart, clearCart } from '../actions/cartActions';
+import axios from 'axios';
+
+interface Product {
+  id: number;
+  title: string;
+  price: number;
+}
+
+interface CartItem extends Product {
+  quantity: number;
+}
+
+const Checkout: React.FC = () => {
+  const cartItems = useSelector((state: RootState) => state.cart.items);
+  const dispatch = useDispatch();
+  const [totalPrice, setTotalPrice] = useState(0);
+
+  useEffect(() => {
+    const fetchPrices = async () => {
+      try {
+        const response = await axios.get('/products');
+        const products: Product[] = response.data;
+        let total = 0;
+        cartItems.forEach((item) => {
+          const product = products.find((p) => p.id === item.id);
+          if (product) {
+            total += product.price * item.quantity;
+          }
+        });
+        setTotalPrice(total);
+      } catch (error) {
+        console.error('Error fetching prices:', error);
+      }
+    };
+
+    fetchPrices();
+  }, [cartItems]);
+
+  useEffect(() => {
+    const updatedTotal = cartItems.reduce(
```

Comment:
Since we can use Redux, this is a perfect candidate for deriving state using a selector. you can learn more about that  [here](https://redux.js.org/usage/deriving-data-selectors#calculating-derived-data-with-selectors).

TLDR; a reusable function that is tied to your global state.